### DISCUSSION TITLE
10.290 Failed to render WebGL canvas

### DISCUSSION THREAD

**Discussion by Rednax141** [felddy/foundryvtt-docker#532] [Q&A] [answered]
Until a few days ago I only used version 8.8. Since I tried the latest stable version I get this error even after a fresh install with just one game system and one new world.

![image](https://user-images.githubusercontent.com/73391271/205465364-0a71307b-5401-46c4-b2ae-48d51e357f5f.png)

I have a pi 4 running portainer and docker to potentially switch between foundry versions. Sometimes I switched between 8.8 and 7.8 and that worked.

I tried enabling the hardware acceleration in chrome on my client but that doesn't change anything. What could be the issue here?

**Comment by felddy** [comment, marked as answer]
This error is probably not related to this Docker container.   
Try checking the official Discord server to see if others have had this issue.
Make sure your game system is compatible with the version of the server you are using.

See:
- https://discord.com/invite/foundryvtt
- https://discord.com/channels/170995199584108546/486930822465716249/917775278292488263

Since [ChatGPT](https://chat.openai.com/chat) is currently all the rage, here is what it had to offer:

>The error `TypeError: Failed to render WebGL canvas. Cannot read properties of undefined (reading 'add')` is typically caused when a web page or application is trying to use WebGL (Web Graphics Library) but is unable to access the required WebGL functions. This can happen for a number of reasons, including:

> - The user's browser does not support WebGL
> - The user's graphics card or drivers do not support WebGL
> - The web page or application has not been properly configured to use WebGL
> - The web page or application is trying to access WebGL functions that do not exist or are not supported by the user's browser or graphics card

>To fix this error, the user can try updating their browser or graphics drivers to the latest version, or the web developer can ensure that the web page or application is properly configured to use WebGL and does not try to access unsupported functions.

**Reply by Rednax141** [reply]
Thanks for your help!

I already asked at the discord and they sent me here. But I kept digging through some more posts on discord. It was a cache problem. I just cleared the cache and everything works fine.

So I guess the learning is: if you host different versions of foundry on the same domain, always check if its the cache first.
